runtime/lua/vim/lsp/util.lua

```diff
@@ -701,19 +701,28 @@ do
     return severity_highlights[severity]
   end
 
-  function M.show_line_diagnostics()
+  function M.get_line_diagnostics()
     local bufnr = api.nvim_get_current_buf()
-    local line = api.nvim_win_get_cursor(0)[1] - 1
+    local linenr = api.nvim_win_get_cursor(0)[1] - 1
+
+    local buffer_diagnostics = M.diagnostics_by_buf[bufnr]
+
+    if not buffer_diagnostics then
+      return {}
+    end
+
+    local diagnostics_by_line = M.diagnostics_group_by_line(buffer_diagnostics)
+    return diagnostics_by_line[linenr]
```

Comment:
not used

Reply:
It is used [here](https://github.com/neovim/neovim/pull/11607/files/618ed87de05a7fc1f7fc475c126d257853948b4c#diff-274652c0882365aea521edadd15ecf0dR166) for getting the diagnostics that might be involved in the code action